src/lib/models/contact-v2.js

```diff
@@ -40,13 +42,41 @@ class Contact extends Model {
     return this._lastName;
   }
 
+  set suffix (suffix) {
+    assertNullableString(suffix);
+    this._suffix = suffix;
+  }
+
+  get suffix () {
+    return this._suffix;
+  }
+
+  set department (department) {
+    assertNullableString(department);
+    this._department = department;
+  }
+
+  get department () {
+    return this._department;
+  }
+
+  set type (type) {
+    assertNullableEnum(type, CONTACT_TYPES);
+    this._type = type;
+  }
+
+  get type () {
+    return this._type;
+  }
+
   /**
-   * Gets the contact's full name including salutation, first name/initial and surname
+   * Gets the contact's full name including title, first name/initial, surname and suffix
    * @return {String}
    */
   get fullName () {
-    const parts = [this._salutation, this._initials || this._firstName, this._lastName];
-    return parts.filter(x => x).join(' ');
+    const parts = [this._title, this._initials || this._firstName, this._lastName];
```

Comment:
This would need to handle initials/middle initials.

I think we would also need to store the `dataSource` property so we could decide how to map the name.